core/src/main/java/feign/template/UriUtils.java

```diff
@@ -153,16 +154,21 @@ private static String encodeChunk(String value, Charset charset, boolean allowRe
     }
 
     byte[] data = value.getBytes(charset);
-    ByteArrayOutputStream encoded = new ByteArrayOutputStream();
-    for (byte b : data) {
-      if (isUnreserved(b) || (isReserved(b) && allowReserved)) {
-        encoded.write(b);
-      } else {
-        /* percent encode the byte */
-        pctEncode(b, encoded);
+    try (ByteArrayOutputStream bos = new ByteArrayOutputStream()) {
+      for (byte b : data) {
+        if (isUnreserved((char) b)) {
+          bos.write(b);
+        } else if (isReserved((char) b) && allowReserved) {
+          bos.write(b);
+        } else {
+          pctEncode(b, bos);
+        }
       }
+      return new String(bos.toByteArray());
```

Comment:
Should we include a charset when creating the string?

Reply:
Sure thing.  Updated.